src/index.js

```diff
@@ -115,10 +115,26 @@ const main = async () => {
     post = await db.collection('posts').findOne({ _id: id })
     res.status(201).send(post)
   })
-  // retreive news and comment sorted by timestamp in reversed order
+  // retreive news and comment sorted by wilson score or timestamp in reversed order
   app.get('/posts', async (req, res) => {
-    const timeline = await db.collection('posts').find().sort({ timestamp: -1 }).toArray()
-    res.json(timeline)
+    let list
+    switch (req.query.params) {
+      case 'score':
+        list = await db.collection('posts').find().sort({ wilson_score: -1 }).toArray()
+        res.json(list)
+        break
+      case 'time':
+        list = await db.collection('posts').find().sort({ timestamp: -1 }).toArray()
+        res.json(list)
+        break
+      case 'downvote':
+        list = await db.collection('posts').find().sort({ downvote: -1 }).toArray()
+        res.json(list)
+        break
+      default:
```

Comment:
We want to sort by score by default. But this will also catch illegal values. You can write `switch (req.query.sortBy || 'score'` to set a default value, and use `default` here to handle illegal values by sending `400`.